podvm/files/etc/systemd/system/process-user-data.service

```diff
@@ -0,0 +1,17 @@
+# One-shot systemd service for running the process-user-data binary
+# It must run before kata-agent.service
+
+[Unit]
+Description=Process user data
+After=cloud-init.target
```

Comment:
Why depend on `cloud-init` here? I think this yields a dependency cycle:

```
Sep 06 07:55:32 localhost.localdomain systemd[1]: process-user-data.service: Found dependency on cloud-final.service/start
Sep 06 07:55:32 localhost.localdomain systemd[1]: process-user-data.service: Found dependency on multi-user.target/start
Sep 06 07:55:32 localhost.localdomain systemd[1]: process-user-data.service: Found dependency on attestation-agent.service/start
Sep 06 07:55:32 localhost.localdomain systemd[1]: process-user-data.service: Found dependency on kata-agent.service/start
Sep 06 07:55:32 localhost.localdomain systemd[1]: process-user-data.service: Found dependency on process-user-data.service/start
```

Reply:
also, if we have a dependency on cloud-init here, we wouldn't see any startup time improvements, would we?